federation/plumbing/queue_server.py

```diff
@@ -0,0 +1,65 @@
+from bluesky_adaptive.utils import extract_event_page
+from bluesky_adaptive.recommendations import NoRecommendation
+
+
+def index_reccomender_factory(
+    *,
+    adaptive_object,
+    sample_index_key,
+    sample_data_key,
+    queue_server,
+    # TODO: Add more sensible defaults for these args.
+    mv_kwargs=None,
+    count_args=(),
+    count_kwargs=None,
+):
+    if mv_kwargs is None:
+        mv_kwargs = {}
+    if count_kwargs is None:
+        count_kwargs = {}
+
+    def callback(name, doc):
+        """Assumes the start doc gives you the sample location,
+        and the event_page gives quality info. The current index is updated at the start
+        But the Agent quality matrix is only updated at tell."""
+        # TODO: Validate the assumptions on formats
+        print(f"callback received {name}")
+
+        if name == "start":
+            current_index = doc[sample_index_key]
+            adaptive_object.tell(x=current_index)
+
+        elif name == "event_page":
+            data = extract_event_page(
+                [
+                    sample_data_key,
+                ],
+                payload=doc["data"],
+            )
+            adaptive_object.tell(y=data)
+
+            try:
+                next_point = adaptive_object.ask(1)
+            except NoRecommendation:
+                queue_server.queue_item_add(None)
+            else:
+                response = queue_server.queue_item_add(
```

Comment:
the `pdf_count` plan will do both the move and the count (and get the meta-data right).